### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by DCY1117**

### CODE REVIEW

File: Assignment4/DCY1117-a180194/task06.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/github/DCY1117/Curso2022-2023-ODKG/blob/master/Assignment4/course_materials/notebooks/Task06.ipynb
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
```
Comment (dgarijo) [inline Assignment4/DCY1117-a180194/task06.py:12]: won't work in python. This is notebook magic

File: Assignment4/DCY1117-a180194/task07.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  for s1, p1, o1 in g.triples((None, RDF.type, s)):
+    print(s1)
+  
+print("SPARQL##########")
+
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    { ?s rdfs:subClassOf ns:Person.
```
Comment (dgarijo) [inline Assignment4/DCY1117-a180194/task07.py:67]: use * or + to retrieve all the subclassof taxonomy

File: Assignment4/DCY1117-a180194/task07.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  for s1, p1, o1 in g.triples((None, RDF.type, s)):
+    for s2, p2, o2 in g.triples((s1, None, None)):
+      print(s2, p2, s)
+
+print("SPARQL##########")
+
+q1 = prepareQuery('''
+  SELECT ?Subject ?prop ?class WHERE { 
+    { ?class rdfs:subClassOf ns:Person.
```
Comment (dgarijo) [inline Assignment4/DCY1117-a180194/task07.py:100]: same as above

File: Assignment4/DCY1117-a180194/task07.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      ?Subject rdf:type ?class.
+      ?Subject ?prop ?o. }
+    UNION
+    { ?Subject rdf:type ns:Person. 
+      ?Subject ?prop ?o 
+      OPTIONAL{ ?Subject rdf:type ?class }}
```
Comment (dgarijo) [inline Assignment4/DCY1117-a180194/task07.py:106]: Optional is incorrect. You are not binding ?class to anything

File: Assignment4/DCY1117-a180194/task08.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  }
+  ''',
+  initNs = { "ns": ns ,"VCARD": VCARD }
+)
+
+for r in g2.query(q2):
+  print(r)
+
+print("Completar campos en el grafo 1############")
+
```
Comment (dgarijo) [inline Assignment4/DCY1117-a180194/task08.py:62]: This is not correct. You should add the trples programatically, not by hand!

File: Assignment4/DCY1117-a180194/task09.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  print(r)
+
+print("Individuos, apodo y nombre de familia grafo 2##############")
+for r in g2.query(q1):
+  print(r)
+
+print("inserta estas coincidencias en g3########")
+g3.add((URIRef('http://data.three.org#SaraJones'), OWL.sameAs, URIRef('http://data.four.org#0001')))
+g3.add((URIRef('http://data.three.org#JohnDoe'), OWL.sameAs, URIRef('http://data.four.org#0005')))
+
```
Comment (dgarijo) [inline Assignment4/DCY1117-a180194/task09.py:50]: It should be done programatically, not by hand

**Review by dgarijo — CHANGES_REQUESTED** [review]

File: Assignment4/DCY1117-a180194/task09.py (hunk 6 above)
Comment (DCY1117) [reply Assignment4/DCY1117-a180194/task09.py:46]: Thanks, changes done.

**Review by dgarijo — APPROVED** [review]

**Comment by dgarijo** [comment]
Small issue in the optional tasks, but the main tasks are correct

**Pull request merged**